blender/LilySurfaceScrapper/CyclesMaterialData.py

```diff
@@ -40,59 +39,66 @@ def createMaterial(self):
-
             if hasattr(texture_node, "color_space"):
                 texture_node.color_space = "COLOR" if map_name == "baseColor" else "NONE"
-            elif map_name == "glossiness":
+            if map_name == "opacity":
+                mat.blend_method = 'BLEND'
+
+        front = back if not front else front # In case just the backside texture was chosen
+
+        def advanced_setup(name, node):
+            if name == "glossiness":
                 invert_node = nodes.new(type="ShaderNodeInvert")
-                links.new(texture_node.outputs["Color"], invert_node.inputs["Color"])
-                links.new(invert_node.outputs["Color"], current_principled.inputs["Roughness"])
-                current_principled
-            elif map_name == "height":
+                links.new(node.outputs["Color"], invert_node.inputs["Color"])
+                links.new(invert_node.outputs["Color"], principled.inputs["Roughness"])
+            elif name == "height":
                 displacement_node = nodes.new(type="ShaderNodeDisplacement")
-                links.new(texture_node.outputs["Color"], displacement_node.inputs["Height"])
+                links.new(node.outputs["Color"], displacement_node.inputs["Height"])
                 links.new(displacement_node.outputs["Displacement"], mat_output.inputs[2])
-            elif map_name == "normal":
+            elif name == "normal":
                 normal_node = nodes.new(type="ShaderNodeNormalMap")
-                links.new(texture_node.outputs["Color"], normal_node.inputs["Color"])
-                links.new(normal_node.outputs["Normal"], current_principled.inputs["Normal"])
-            else:
-                links.new(texture_node.outputs["Color"], current_principled.inputs[__class__.input_tr[map_name]])
-
-            if map_name == "opacity":
-                mat.blend_method = 'BLEND'
-            
-            if displacement_node is not None and normal_node is not None:
-                    links.new(normal_node.outputs["Normal"], displacement_node.inputs["Normal"])
+                links.new(node.outputs["Color"], normal_node.inputs["Color"])
+                links.new(normal_node.outputs["Normal"], principled.inputs["Normal"])
 
+        if not back: # If there is no item in the back dictionary
+            for name, node in front.items():
+                if __class__.input_tr.get(name):
+                    links.new(node.outputs["Color"], principled.inputs[__class__.input_tr[name]])
+                else:
+                    advanced_setup(name, node)
+        else:
+            geometry_node = nodes.new("ShaderNodeNewGeometry")
+            def setup(name, front, back, mix):
+                links.new(geometry_node.outputs["Backfacing"], mix.inputs[0])
+                links.new(front.outputs["Color"], mix.inputs[1])
+                links.new(back.outputs["Color"], mix.inputs[2])
+                if __class__.input_tr.get(name):
```

Comment:
This is a duplication of lines 84-87 with just the substitution `node` <- `mix` ;)

Reply:
https://github.com/ogierm/LilySurfaceScrapper/commit/c59603ccabf8f3f4e3d9e19862b8537f4a84d89d#r34949766